### PR TITLE
Add linting config and fix issues.

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title. Prep work for knative-sandbox/.github#13 to land in sandbox.

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  skip-dirs:
+    - pkg/client
+    - pkg/internal/thirdparty
+
+linters:
+  enable:
+    - unconvert
+    - prealloc
+  disable:
+    - errcheck
```
Comment (mattmoor) [inline .golangci.yaml:16]: Add trailing newline:
```suggestion
    - errcheck

```

**Review by mattmoor — COMMENTED** [review]
/lgtm

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**